test/base/test_objective.py

```diff
@@ -160,3 +165,40 @@ def rel_err(eps_):
         objective.check_grad_multi_eps([theta], multi_eps=multi_eps)
     assert result_multi_eps['rel_err'].squeeze() == \
         min(rel_err(_eps) for _eps in multi_eps)
+
+
+def test_aesara(max_sensi_order, integrated):
+    prob = rosen_for_sensi(max_sensi_order,
+                           integrated, [0, 1])
+
+    base_objective = AesaraLogProbability(pypesto.Problem(
+        prob['obj'], -np.inf * np.ones_like(prob['x']),
+        +np.inf * np.ones_like(prob['x'])),
+        beta=-1
+    )
+    x = aet.specify_shape(aet.vector('x'), (2,))
+    # compose with sinh
+    fun = base_objective(aet.sinh(x))
+
+    f = aesara.function([x], fun)
+    kwargs = {'fun': lambda z: float(f(z))}
+    if max_sensi_order > 0:
+        g = aesara.function([x], aesara.grad(fun, [x]))
+        kwargs['grad'] = lambda z: g(z)[0]
+    if max_sensi_order > 1:
+        h = aesara.function([x], aesara.gradient.hessian(fun, [x]))
+        kwargs['hess'] = lambda z: h(z)[0]
+
+    obj = pypesto.Objective(**kwargs)
+
+    assert obj(prob['x']) == prob['obj'](np.sinh(prob['x']))
+
+    if max_sensi_order > 0:
+        df = obj.check_grad(prob['x'])
+        assert (df.abs_err < 1e-6).all() & (df.rel_err < 1e-6).all()
+
+    if max_sensi_order > 1:
+        hess = obj(prob['x'], sensi_orders=(2,))
+        assert np.isclose(hess[0, 0], -468.08047745752054)
+        assert np.isclose(hess[1, 1], 752.4391382167262)
```

Comment:
an analytic expression would be more credible, but should be fine ;)

Reply:
added analytical solutions